dotnet/src/Connectors/Connectors.OpenAIV2/Core/ClientCore .AudioToText.cs

```diff
@@ -0,0 +1,119 @@
+/// </summary>
+internal partial class ClientCore
+{
+    /// <summary>
+    /// Generates an image with the provided configuration.
+    /// </summary>
+    /// <param name="input">Input audio to generate the text</param>
+    /// <param name="executionSettings">Audio-to-text execution settings for the prompt</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Url of the generated image</returns>
+    internal async Task<IReadOnlyList<TextContent>> GetTextFromAudioContentsAsync(
+        AudioContent input,
+        PromptExecutionSettings? executionSettings,
+        CancellationToken cancellationToken)
+    {
+        if (!input.CanRead)
+        {
+            throw new ArgumentException("The input audio content is not readable.", nameof(input));
+        }
+
+        OpenAIAudioToTextExecutionSettings? audioExecutionSettings = OpenAIAudioToTextExecutionSettings.FromExecutionSettings(executionSettings);
+        AudioTranscriptionOptions? audioOptions = AudioOptionsFromExecutionSettings(executionSettings);
+
+        Verify.ValidFilename(audioExecutionSettings?.Filename);
+
+        var memoryStream = new MemoryStream(input.Data!.Value.ToArray());
+
+        AudioTranscription responseData = (await RunRequestAsync(() => this.Client.GetAudioClient(this.ModelId).TranscribeAudioAsync(memoryStream, audioExecutionSettings?.Filename, audioOptions)).ConfigureAwait(false)).Value;
+
+        return [new(responseData.Text, this.ModelId, metadata: GetResponseMetadata(responseData))];
+    }
+
+    /// <summary>
+    /// Converts <see cref="PromptExecutionSettings"/> to derived <see cref="OpenAIAudioToTextExecutionSettings"/> type.
+    /// </summary>
+    /// <param name="executionSettings">Instance of <see cref="PromptExecutionSettings"/>.</param>
+    /// <returns>Instance of <see cref="OpenAIAudioToTextExecutionSettings"/>.</returns>
+    private static AudioTranscriptionOptions? AudioOptionsFromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        if (executionSettings is null)
+        {
+            return new AudioTranscriptionOptions();
+        }
+
+        if (executionSettings is OpenAIAudioToTextExecutionSettings settings)
+        {
+            return new AudioTranscriptionOptions
+            {
+                Granularities = ConvertToAudioTimestampGranularities(settings.Granularities),
+                Language = settings.Language,
+                Prompt = settings.Prompt,
+                Temperature = settings.Temperature
+            };
+        }
+
+        var json = JsonSerializer.Serialize(executionSettings);
+
+        var openAIExecutionSettings = JsonSerializer.Deserialize<OpenAIAudioToTextExecutionSettings>(json, JsonOptionsCache.ReadPermissive);
+
+        if (openAIExecutionSettings is not null)
```

Comment:
If in this case is possibility openAIExecutionSettings will be null? I think you can remove null check and suppress.